From 58d9a11c9f96e5894c4c28b547f690e8761d5727 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Fri, 10 Jul 2020 09:47:07 +0800
Subject: [PATCH] policy/modules/kernel/domain: allow domain to map generic
 files in /etc

Allow domain to map /etc/passwd, /etc/group

Fixes:
avc:  denied  { map } for  pid=242 comm="systemd-tmpfile"
path="/etc/passwd" dev="sdb2" ino=793
scontext=system_u:system_r:systemd_tmpfiles_t:s0-s15:c0.c1023
tcontext=system_u:object_r:etc_t:s0 tclass=file permissive=0

avc:  denied  { map } for  pid=242 comm="systemd-tmpfile"
path="/etc/group" dev="sdb2" ino=32952
scontext=system_u:system_r:systemd_tmpfiles_t:s0-s15:c0.c1023
tcontext=system_u:object_r:etc_t:s0 tclass=file permissive=0

avc:  denied  { map } for  pid=238 comm="alsactl" path="/etc/passwd"
dev="sdb2" ino=793 scontext=system_u:system_r:alsa_t:s0-s15:c0.c1023
tcontext=system_u:object_r:etc_t:s0 tclass=file permissive=0

avc:  denied  { map } for  pid=350 comm="login" path="/etc/passwd"
dev="sdb2" ino=793
scontext=system_u:system_r:local_login_t:s0-s15:c0.c1023
tcontext=system_u:object_r:etc_t:s0 tclass=file permissive=0

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/kernel/domain.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/kernel/domain.te b/policy/modules/kernel/domain.te
index 7e5d2b458..cc88c5eea 100644
--- a/policy/modules/kernel/domain.te
+++ b/policy/modules/kernel/domain.te
@@ -113,6 +113,8 @@ files_list_root(domain)
 # Yocto/oe-core use some var volatile links
 files_read_var_symlinks(domain)
 
+files_map_etc_files(domain)
+
 ifdef(`hide_broken_symptoms',`
 	# This check is in the general socket
 	# listen code, before protocol-specific
-- 
2.17.1

